internal/hud/webview/convert.go

```diff
@@ -144,7 +144,11 @@ func resourceInfoView(mt *store.ManifestTarget) ResourceInfoView {
 		dc := mt.Manifest.DockerComposeTarget()
 		dcState := mt.State.DCRuntimeState()
 		return NewDCResourceInfo(dc.ConfigPaths, dcState.Status, dcState.ContainerID, dcState.Log(), dcState.StartTime)
-	} else {
+	}
```

Comment:
I updated this in my PR -- [now it's a type switch on mt.State](https://github.com/windmilleng/tilt/blob/master/internal/store/engine_state.go#L603) (with a dummy state for LocalResource), which seems cleaner. (and i still return an empty k8sRuntimeState as my base case, instead of panicking -- technically less correct but also less likely to loudly break ppl? idk, I could go either way but @dbentley leaned towards the non-panic so I did that.

Reply:
I wish Go had tagged unions, then the compiler could enforce that we check all possible values for an enum when we switch over them. In the meantime I like the panic solution since it makes it more obvious where we need to change the code next time we add a resource type, and it's basically impossible for us to not notice it heh.

Reply:
fair nough. want to change my stupid return statement back to a panic for consistency?

Reply:
Sure, will do in a separate PR.